presto-product-tests/src/main/java/io/prestosql/tests/hive/acid/TestInsertOnlyAcidTableRead.java

```diff
@@ -0,0 +1,131 @@
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.sql.SQLException;
+
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+import static io.prestosql.tempto.query.QueryExecutor.query;
+import static io.prestosql.tests.TestGroups.STORAGE_FORMATS;
+import static io.prestosql.tests.hive.acid.HiveHelper.canRunAcidTests;
+import static io.prestosql.tests.utils.QueryExecutors.onHive;
+import static java.util.Locale.ENGLISH;
+
+public class TestInsertOnlyAcidTableRead
+        extends ProductTest
+{
+    @Test(groups = STORAGE_FORMATS, dataProvider = "insertOnlyAcidTableTypes")
+    public void testSelectFromInsertOnlyAcidTable(InsertOnlyAcidTableType type)
+            throws SQLException
+    {
+        if (!canRunAcidTests()) {
+            return;
+        }
+
+        String tableName = "insert_only_acid_table" + type.name().toLowerCase(ENGLISH);
+        createTable(tableName, type.isPartitioned());
+
+        try {
+            onHive().executeQuery(
+                    "INSERT OVERWRITE TABLE " + tableName + getHivePartitionString(type.isPartitioned()) + " select 1");
+
+            String selectFromOnePartitionsSql = "SELECT col FROM " + tableName + getPrestoPredicate(type.isPartitioned()) + " ORDER BY COL";
+            QueryResult onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsOnly(row(1));
+
+            onHive().executeQuery(
+                    "INSERT INTO TABLE " + tableName + getHivePartitionString(type.isPartitioned()) + " select 2");
+            onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsExactly(row(1), row(2));
+
+            onHive().executeQuery(
+                    "INSERT OVERWRITE TABLE " + tableName + getHivePartitionString(type.isPartitioned()) + " select 3");
+            onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsOnly(row(3));
+        }
+        finally {
+            onHive().executeQuery("DROP TABLE " + tableName);
+        }
+    }
+
+    private String getHivePartitionString(boolean isPartitioned)
+    {
+        if (!isPartitioned) {
+            return "";
+        }
+
+        return " PARTITION (part_col=2) ";
+    }
+
+    private String getPrestoPredicate(boolean isPartitioned)
+    {
+        if (!isPartitioned) {
+            return "";
+        }
+
+        return " WHERE part_col = 2 ";
+    }
+
+    private void createTable(String tableName, boolean isPartitioned)
+    {
+        StringBuilder builder = new StringBuilder()
+                .append("CREATE TABLE IF NOT EXISTS ")
+                .append(tableName)
+                .append(" (col INT) ");
+        if (isPartitioned) {
+            builder.append("PARTITIONED BY (part_col INT) ");
```

Comment:
Use string concatenation instead of StringBuilder.
Use elvis for defining the partitionign column